include/ic_common.h

```diff
@@ -1,7 +1,16 @@
 #pragma once
 
-#include <functional>
+#include "ic_log.h"
 
-namespace IC {
-    template <class Ret, class... Args> using Func = std::function<Ret(Args...)>;
-}
+#define IC_ASSERT(x, ...)                                                                                              \
```

Comment:
Just wanna quickly note here that I did have these as one-liners, but the formatter expanded them out. We can talk about if we wanna keep that setting(s?) or not.